packages/flutter/test/widgets/scrollable_helpers_test.dart

```diff
@@ -590,4 +606,77 @@ void main() {
     expect(find.text('The cape as red as blood'), findsOneWidget);
     expect(find.text('The hair as yellow as corn'), findsOneWidget);
   });
+
+  testWidgets('Invoking a ScrollAction when notificationContext is null does not cause an exception.', (WidgetTester tester) async {
+    const List<LogicalKeyboardKey> keysWithModifier = <LogicalKeyboardKey>[
+      LogicalKeyboardKey.arrowDown, LogicalKeyboardKey.arrowUp,
+    ];
+    const List<LogicalKeyboardKey> keys = <LogicalKeyboardKey>[
+      ...keysWithModifier,
+      LogicalKeyboardKey.pageDown, LogicalKeyboardKey.pageUp,
+    ];
+    final ScrollController controller = ScrollController();
+    addTearDown(controller.dispose);
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(platform: TargetPlatform.fuchsia),
+        home: PrimaryScrollController(
+          controller: controller,
+          child: Focus(
+            autofocus: true,
+            child: _NoNotificationContextScrollable(
+              controller: controller,
+              viewportBuilder: (BuildContext context, ViewportOffset offset) => Viewport(
+                offset: offset,
+                slivers: List<Widget>.generate(
+                  20,
+                  (int index) => SliverToBoxAdapter(
+                    child: SizedBox(
+                      key: ValueKey<String>('Box $index'),
+                      height: 50.0,
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Verify the initial scroll offset.
+    await tester.pumpAndSettle();
+    expect(controller.position.pixels, equals(0.0));
+    expect(
+      tester.getRect(find.byKey(const ValueKey<String>('Box 0'), skipOffstage: false)),
+      equals(const Rect.fromLTRB(0.0, 0.0, 800.0, 50.0)),
+    );
+
+    for (final LogicalKeyboardKey key in keys) {
+      // We exclude the modifier keys here for web testing since default web shortcuts
+      // do not use a modifier key with arrow keys for ScrollActions.
+      if (!kIsWeb && keysWithModifier.contains(key)) {
+        await tester.sendKeyDownEvent(modifierKey);
+      }
+
+      // Invoking the ScrollAction would previously lead to a type error caused by an
+      // unsafe non-null assertion on the notificationContext, let's make sure it
+      // doesn't happen anymore.
```

Comment:
Instead of describing the bug in the test code, we usually just add an issue link.

```suggestion
```